backend/routes/room.js

```diff
@@ -8,21 +8,65 @@ const users = require('../models/User.model');
 const shortid = require('shortid');
 var io = require('../index');
 const auth = require('../middleware/auth');
-
+const cron = require('cron').CronJob;
 //Create a new room
 /* TODO: Modify the function rollback changes on failure*/
+
+//Run this function as a chron job every x hours
+function cleanDB() {
+  const now = new Date();
+  // console.log(now);
+  rooms.find({ lastreq: { $lt: now } }, (err, room) => {
+    room.map((val, index) => {
+      const data = val.toObject();
+      //Getting the time since last join call by a user in room (in hours)
+      let timeSinceLastCall = (now - data.lastreq) / 36e5;
+      let clearoutTime = 12; // Set the time (in hours) after which the guest and online array need to be cleaned
+      console.log(timeSinceLastCall);
+      if (timeSinceLastCall > clearoutTime) {
+        rooms.updateOne(
+          { roomName: data.roomName },
+          { $set: { onlineSimple: [], guests: [] } },
+          (err, res) => {
+            if (err) console.log(err);
+          }
+        );
+      }
+    });
+  });
+}
+
+//This part runs the CronJob
+//Currently runs every 4 hours
+var job = new cron(
+  '0 1,13 * * *',
+  function () {
+    console.log('Running CronJob');
+    cleanDB();
+  },
+  null,
+  true,
+  'America/Los_Angeles'
```

Comment:
Is this time zone?
If yes then why America?

Reply:
Indian time zone showed as invalid entry. Hence chose the default one.

Reply:
Timezone should be same as that of server, if not then it'll cause issues.

Can you check the documentation of package for corresponding Indian Timezone.

Reply:
Okay will do